unit-tests/src/main/scala/scala/ArrayCopySuite.scala

```diff
@@ -0,0 +1,435 @@
+          arr(3) == 3 && arr(4) == 4 && arr(5) == 5 &&
+          arr(6) == 6 && arr(7) == 7 && arr(8) == 8 &&
+          arr(9) == 9
+    )
+  }
+
+  test("array[Int]: copy to another array") {
+    init(arr)
+    scala.Array.copy(arr, 0, arr2, 1, 10)
+    assert(
+        arr2(0) == 0 && arr2(1) == 0 && arr2(2) == 1 &&
+          arr2(3) == 2 && arr2(4) == 3 && arr2(5) == 4 &&
+          arr2(6) == 5 && arr2(7) == 6 && arr2(8) == 7 &&
+          arr2(9) == 8 && arr2(10) == 9 && arr2(11) == 0
+    )
+  }
+
+  test("array[Int]: copy zero elements from empty array") {
+    init(arr2)
+    scala.Array.copy(arrEmpty, 0, arr2, 5, 0)
+    assert(
+        arr2(0) == 0 && arr2(1) == 1 && arr2(2) == 2 &&
+          arr2(3) == 3 && arr2(4) == 4 && arr2(5) == 5 &&
+          arr2(6) == 6 && arr2(7) == 7 && arr2(8) == 8 &&
+          arr2(9) == 9 && arr2(10) == 10 && arr2(11) == 11
+    )
+  }
+
+  test("array[Int]: copy to self without overlap (1/2)") {
+    init(arr)
+    scala.Array.copy(arr, 0, arr, 5, 5)
+    assert(
+        arr(0) == 0 && arr(1) == 1 && arr(2) == 2 && arr(3) == 3 &&
+          arr(4) == 4 && arr(5) == 0 && arr(6) == 1 && arr(7) == 2 &&
+          arr(8) == 3 && arr(9) == 4
+    )
+  }
+
+  test("array[Int]: copy to self without overlap (2/2)") {
+    init(arr)
+    scala.Array.copy(arr, 6, arr, 4, 2)
+    assert(
+        arr(0) == 0 && arr(1) == 1 && arr(2) == 2 && arr(3) == 3 &&
+          arr(4) == 6 && arr(5) == 7 && arr(6) == 6 && arr(7) == 7 &&
+          arr(8) == 8 && arr(9) == 9
+    )
+  }
+
+  test("array[Int]: copy to self with overlap and backward copy") {
+    init(arr)
+    scala.Array.copy(arr, 0, arr, 2, 6)
+    assert(
+        arr(0) == 0 && arr(1) == 1 && arr(2) == 0 && arr(3) == 1 &&
+          arr(4) == 2 && arr(5) == 3 && arr(6) == 4 && arr(7) == 5 &&
+          arr(8) == 8 && arr(9) == 9
+    )
+  }
+
+  test("array[Int]: copy to self with overlap and forward copy") {
+    init(arr)
+    scala.Array.copy(arr, 2, arr, 0, 6)
+    assert(
+        arr(0) == 2 && arr(1) == 3 && arr(2) == 4 && arr(3) == 5 &&
+          arr(4) == 6 && arr(5) == 7 && arr(6) == 6 && arr(7) == 7 &&
+          arr(8) == 8 && arr(9) == 9
+    )
+  }
+
+  test("array[Int]: from is null") {
+    assertThrows[java.lang.NullPointerException] {
+      scala.Array.copy(null, 0, arr2, 5, 2)
+    }
+  }
+
+  test("array[Int]: to is null") {
+    assertThrows[java.lang.NullPointerException] {
+      scala.Array.copy(arr, 0, null, 5, -2)
+    }
+  }
+
+  test("array[Int]: length is negative") {
+    assertThrows[java.lang.IndexOutOfBoundsException] {
+      scala.Array.copy(arr, 0, arr2, 5, -1)
+    }
+  }
+
+  test("array[Int]: toPos + len > to.length") {
+    assertThrows[java.lang.IndexOutOfBoundsException] {
+      scala.Array.copy(arr, 0, arr2, 5, 10)
+    }
+  }
+
+  test("array[Int]: fromPos + len > from.length") {
+    assertThrows[java.lang.IndexOutOfBoundsException] {
+      scala.Array.copy(arr, 5, arr2, 5, 10)
```

Comment:
I think `toPos + len` should be `< arr2.length`
similarly in `"array[Int]: fromPos is negative"` test case